packages/core/ui-settings/core-ui-settings-browser-internal/src/ui_settings_client_common.ts

```diff
@@ -100,6 +100,22 @@ You can use \`IUiSettingsClient.get("${key}", defaultValue)\`, which will just r
     return await this.update(key, null);
   }
 
+  allowlist(keys: string[]) {
+    keys.forEach((key) => {
+      if (this.defaults.hasOwnProperty(key)) {
+        this.defaults[key].allowlisted = true;
+      }
+      if (this.cache.hasOwnProperty(key)) {
+        this.cache[key].allowlisted = true;
+      }
+    });
+  }
+
+  allowlistAll() {
+    Object.entries(this.defaults).forEach(([key, value]) => (value.allowlisted = true));
+    Object.entries(this.cache).forEach(([key, value]) => (value.allowlisted = true));
+  }
```

Comment:
> to have a new API along the lines of `coreSetup.uiSettings.setAllowlist([...]);` which can only be called once (or it throws an error).

Based on my understanding, we want to allowlist the settings for all serverless projects from the serverless plugin, the setttings for the security project from serverless security plugin, etc. (@clintandrewhall can you confirm that this is still the plan?). Therefore, we might need to call `setAllowlist()` two times (once from the serverless plugin and once from the serverless plugin for the specific project). Would there be an issue with this?

Alternatively, if we want to only allow calling it once, I suppose we could only call it from the serverless project plugins (i.e. the serverless security plugin allowlisting the general settings + the security-specific settings).

> Then we'd need to throw an error if someone tried to `set` or `get` a setting that wasn't allowed, or if they tried to do the same via the HTTP APIs as mentioned in my other comment.

I'm wondering if this might not break something since there might be settings that we don't want to display in the Advanced settings app, but are consumed by other plugins in Kibana. For example, I see that there is a setting [theme:version](https://github.com/elastic/kibana/blob/5aee5da843fc1e7747b66bb45e8ebf7a808c144d/packages/core/ui-settings/core-ui-settings-server-internal/src/settings/theme.ts#L60) which has the `readonly` flag set to true and so it's not displayed in the UI but is accessed by some places. If we don't allow getting settings that are not allowlisted, this setting would also not be accessible. Probably the same applies for settings with the `isCustom` flag as they are also [hidden in the UI](https://github.com/elastic/kibana/blob/5aee5da843fc1e7747b66bb45e8ebf7a808c144d/src/plugins/advanced_settings/public/management_app/settings_helper.ts#L27).

Overall, I thought it would be okay for other places in Kibana to `get` and `set` settings that are not allowlisted as long as they are not displayed in the UI so that the user can't change them. Would the user still be able to change them if they are just hidden?

Reply:
The third option would be to provide a list of settings from a package-- in this PR, `ALL_ALLOWLIST_SETTINGS`, and each solution sets the entire list one time from their setup contract.  I don't see a reason why this API can't be called multiple times, but I also see the value in calling it only once.